calyx/src/passes/sharing_components.rs

```diff
@@ -92,6 +83,11 @@ impl<T: ShareComponents> Visitor for T {
             .map(|cell| (cell.clone_name(), cell.borrow().prototype.clone()))
             .collect();
 
+        // These two options produce different ordering, which affects which
+        // component's name is used.
```

Comment:
Dangling comment here. Which line is this supposed to refer to?

Reply:
In your PR you had 
1)
```
let cells_by_type: HashMap<&ir::CellType, Vec<&ir::Id>> =
            cell_type.iter().map(|(k, v)| (v, k)).into_group_map();
```
instead of 
2)
```
let mut cells_by_type: HashMap<ir::CellType, Vec<ir::Id>> =
            HashMap::new();
        for cell in cells {
            cells_by_type
                .entry(cell.borrow().prototype.clone())
                .and_modify(|v| v.push(cell.clone_name()))
                .or_insert_with(|| vec![cell.clone_name()]);
        }
```
(You also used the name `cell_type` instead of `id_to_type`). 
I applied these changes, and found that  1 generates nicer looking code, but I found that the timing difference is negligible, and 1) generates sharing that (I think) is correct, but it just names differently (so if a and b can be shared, it renames both to b while the test cases rename to a). So I just went with 2).